File: Beecrowd/PulaSapo/Jogo.java
/**
 * The "Jogo" class represents a game that consists of a frog character jumping over obstacles.
 */
package PulaSapo;

import java.util.ArrayList;

public class Jogo {
    private int numObstaculos;
    private Sapo sapo;
    private ArrayList<Obstaculo> obstaculos = new ArrayList<Obstaculo>();

    Jogo(int sapoJumpHeight, int numObstaculos){
        this.sapo = new Sapo(sapoJumpHeight);
        this.numObstaculos = numObstaculos;
    }

    //*getters */

    public int getNumObstaculos() {
        return numObstaculos;
    }

    public Sapo getSapo() {
        return sapo;
    }

    //*funções */

    /**
     * The function adds an obstacle of a given size to a list of obstacles.
     * 
     * @param obstacleSize The parameter "obstacleSize" represents the size of the obstacle that is
     * being added.
     */
    public void addObstaculo(int obstacleSize){
        obstaculos.add(new Obstaculo(obstacleSize));
    }

    /**
     * The function checks if the game is over by iterating through a list of obstacles and determining
     * if the player's character can jump over each obstacle.
     * 
     * @return The method is returning a boolean value. If the player wins the game, it returns true.
     * If the player loses the game, it returns false.
     */
    public boolean GameOver(){
        this.sapo.setHeight(obstaculos.get(0).getSize());
        for(int i=1;i<obstaculos.size();i++){
            if(sapo.jump(obstaculos.get(i)))
                continue;
            else{
                System.out.println("GAME OVER");
                return false;
            }
        }
        System.out.println("YOU WIN");
        return true;
    }
}

File: Beecrowd/PulaSapo/Jogo.py
from Sapo import Sapo
from Obstaculo import Obstaculo


# The class "Jogo" represents a game where a player controls a frog to jump over obstacles, and the
# class contains methods to add obstacles, check if the player has successfully jumped over all
# obstacles, and get information about the frog and obstacles.
class Jogo:
    __Obstaculos = []

    def __init__(self, sapoJumpHeight, numObstaculos):
        self.__sapo = Sapo(sapoJumpHeight)
        self.__numObstaculos = numObstaculos

    # * getters

    def getSapo(self) -> Sapo:
        return self.__sapo

    def getObstaculo(self, index) -> Obstaculo:
        return self.__Obstaculos[index]

    def getNumObstaculos(self) -> int:
        return self.__numObstaculos
    

    # * funções

    def addObstaculo(self, ObstacleHeight) -> None:
        """
        The function adds an obstacle with a specified height to a list of obstacles.
        
        :param ObstacleHeight: The parameter "ObstacleHeight" represents the height of the obstacle that
        you want to add
        """
        self.__Obstaculos.append(Obstaculo(ObstacleHeight))

    def GameOver(self) -> bool:
        """
        The function checks if the player has successfully jumped over all obstacles and returns True if
        they have, otherwise it returns False and prints "GAME OVER".
        :return: a boolean value. If the player wins the game, it returns True. If the player loses the
        game, it returns False.
        """
        self.__sapo.setHeight(self.__Obstaculos[0].getSize())
        for i in range(1,len(self.__Obstaculos)):
            if self.__sapo.jump(self.__Obstaculos[i]):
                continue
            else:
                print('GAME OVER')
                return False
        
        print('YOU WIN')
        return True